feather/protocol/src/packets/server/play.rs

```diff
@@ -1179,6 +1181,163 @@ packets! {
+            "generic:attack_knockback" => EntityAttributeKind::AttackKnockback,
+            "generic:flying_speed" => EntityAttributeKind::FlyingSpeed,
+            "generic:armor" => EntityAttributeKind::Armor,
+            "generic:armor_toughness" => EntityAttributeKind::ArmorToughness,
+            "generic:luck" => EntityAttributeKind::Luck,
+            "horse:jump_strength" => EntityAttributeKind::HorseJumpStrength,
+            "zombie:spawn_reinforcements" => EntityAttributeKind::ZombieSpawnReinforcements,
+            _ => return Err(anyhow::anyhow!("invalid entity property key: '{}'", key)),
+        })
+    }
+
+    pub fn id(&self) -> String {
+        String::from(match self {
+            EntityAttributeKind::MaxHealth => "generic:max_health",
+            EntityAttributeKind::FollowRange => "generic:follow_range",
+            EntityAttributeKind::KnockbackResistance => "generic:knockback_resistance",
+            EntityAttributeKind::MovementSpeed => "generic:movement_speed",
+            EntityAttributeKind::AttackDamage => "generic:attack_damage",
+            EntityAttributeKind::AttackSpeed => "generic:attack_speed",
+            EntityAttributeKind::AttackKnockback => "generic:attack_knockback",
+            EntityAttributeKind::FlyingSpeed => "generic:flying_speed",
+            EntityAttributeKind::Armor => "generic:armor",
+            EntityAttributeKind::ArmorToughness => "generic:armor_toughness",
+            EntityAttributeKind::Luck => "generic:luck",
+            EntityAttributeKind::HorseJumpStrength => "horse:jump_strength",
+            EntityAttributeKind::ZombieSpawnReinforcements => "zombie:spawn_reinforcements",
+        })
+    }
+
+    pub fn max_value(&self) -> f32 {
+        match &self {
+            EntityAttributeKind::MaxHealth => 1024.0,
+            EntityAttributeKind::FollowRange => 2048.0,
+            EntityAttributeKind::KnockbackResistance => 1.0,
+            EntityAttributeKind::MovementSpeed => 1024.0,
+            EntityAttributeKind::AttackDamage => 2048.0,
+            EntityAttributeKind::AttackSpeed => 1024.0,
+            EntityAttributeKind::AttackKnockback => 5.0,
+            EntityAttributeKind::FlyingSpeed => 1024.0,
+            EntityAttributeKind::Armor => 30.0,
+            EntityAttributeKind::ArmorToughness => 20.0,
+            EntityAttributeKind::Luck => 1024.0,
+            EntityAttributeKind::HorseJumpStrength => 2.0,
+            EntityAttributeKind::ZombieSpawnReinforcements => 1.0,
+        }
+    }
+
+    pub fn min_value(&self) -> f32 {
+        if let EntityAttributeKind::Luck = &self {
+            return -1024.0;
+        }
+
+        0.0
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct AttributeModifier {
+    pub uuid: Uuid,
+    pub amount: f32,
+    pub operation: OperationKind,
+}
+
+#[derive(Debug, Clone)]
+pub struct EntityProperty {
+    pub attribute: EntityAttributeKind,
+    pub modifiers: Vec<AttributeModifier>,
+    pub value: f32,
```

Comment:
This needs to be between `attribute` and `modifiers` when you implement this struct via the `packets!` macro because the declaration order needs to match the protocol order in that case.